lib/alerts/survey_response_alert_manager.rb

```diff
@@ -9,21 +9,33 @@ def create_alerts
     @answers.each do |answer|
       next unless answer_meets_alert_conditions?(answer)
       next if alert_already_exists?(answer)
-      details = {
-        question: answer.question.question_text,
-        answer: answer.answer_text,
-        followup: answer.follow_up_answer_text
-      }
       alert = Alert.create(type: 'SurveyResponseAlert',
                            message: alert_message(answer),
                            user_id: answer.user.id,
                            subject_id: answer.question.id,
-                           details: details,
+                           details: details(answer),
                            target_user_id: alert_recipient&.id)
       alert.email_target_user
     end
   end
 
+  def details(answer)
+    {
+      question: answer.question.question_text,
+      answer: answer.answer_text,
+      followup: answer.follow_up_answer_text,
+      source: self.class.source(answer)
+    }
+  end
+
+  def self.source(answer)
+    question_group = answer.answer_group.question_group
+    survey_names = question_group.surveys.map(&:name).join(' .')
```

Comment:
The idea here is to show all the surveys that this alert might have been generated from?

Reply:
Yes. I wanted to display both Survey(s) and question group, to have all the information.